src/webpack/plugins/pluginExecutor.ts

```diff
@@ -0,0 +1,43 @@
+import {getRootDirectory} from '../../utils/getRootDirectory';
+
+import webpack = require('webpack');
+import {BasePlugin} from './BasePlugin';
+
+type Plugin = BasePlugin<any> | webpack.Plugin;
+
+/**
+ * Executes all runnable plugins without having to do a full build.  Standard webpack plugins will be ignored.
+ *
+ * Will extract code generation plugins from the main webpack config, and then invoke those plugins from "manually",
+ * without using webpack.
+ */
+export async function executeAllPlugins(action?: string) {
```

Comment:
As above, should `action` be required here?

Reply:
Let me give some more thought to this one.